File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.py
class Solution:
    def check(self, nums: List[int]) -> bool:
        n = len(nums)
        x = 0

        if nums[0] < nums[n - 1]:
            x += 1

        for i in range(1, n):
            if (nums[i - 1] <= nums[i]) and x < 2:
                continue
            if nums[i - 1] > nums[i]:
                x += 1
            if x == 2:
                return False

        return True

File: 1878-check-if-array-is-sorted-and-rotated/check-if-array-is-sorted-and-rotated.js
/**
 * @param {number[]} nums
 * @return {boolean}
 */
var check = function (nums) {
    const n = nums.length;
    let count = 0;

    for (let i = 1; i < n; i++) {
        if (nums[i - 1] > nums[i]) {
            count++;
        }
    }

    if ((count > 1) || (count === 1 && nums[0] < nums[n - 1])) {
        return false;
    }

    return true;
};